libs/esp32devmode/src/esp32devmode.erl

```diff
@@ -0,0 +1,314 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2023 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(esp32devmode).
+
+-export([
+    start_dev_mode/0,
+    start_repl/1,
+    start_network/0,
+    handle_req/3,
+    erase_net_config/0,
+    save_net_config/2
+]).
+
+-record(nc_state, {socket, pending_pid, pending_ref}).
+
+-define(DEFAULT_AP_SSID, <<"AtomVM-ESP32">>).
+-define(DEFAULT_AP_PSK, <<"esp32default">>).
+-define(DEFAULT_CONSOLE_PORT, 2323).
+-define(DEFAULT_WEB_SERVER_PORT, 8080).
+
+start_dev_mode() ->
+    avm_pubsub:start(default_pubsub),
+    spawn(?MODULE, fun maybe_start_network/0, []),
```

Comment:
This call to spawn raises a `badarg`.  The second parameter needs to be an atom, and in addition, `maybe_start_network` would need to be exported.

A better solution would be:
```
spawn(fun maybe_start_network/0),
```

and the function can remain private.

Reply:
my mistake during last round of fixes, fixed.